packages/material-ui/src/StepLabel/StepLabel.js

```diff
@@ -96,19 +97,29 @@ function StepLabel(props) {
       )}
       {...other}
     >
-      {icon && (
+      {(icon || StepIconComponent) && (
         <span
           className={classNames(classes.iconContainer, {
             [classes.alternativeLabel]: alternativeLabel,
           })}
         >
+        {StepIconComponent ? (
+          <StepIconComponent
+            completed={completed}
+            active={active}
+            error={error}
+            icon={icon}
+            {...StepIconProps}
+          />
+        ) : (
           <StepIcon
             completed={completed}
             active={active}
             error={error}
             icon={icon}
             {...StepIconProps}
           />
```

Comment:
The same props are applied to both paths. Why not just let `StepIconComponent` default to `StepIcon`?

Reply:
Beat me to it. 😄

Reply:
will do that